### ISSUE TITLE
Pre populated row using counter

### ISSUE THREAD

**Issue by brad-klosterman** [Rhyzz/repeatable-fields#19]
Hi, thank you for this. 

I am storing user meta and then populating it in the row on page load. When the meta in the row is pre-populated on the page and then saved only one value is saved. When I add rows they save correctly. When those rows are pre-populated again on the page load and then saved again it only saves the last field. 

`<form id="pfile" action="" method="POST">	
                            <div class="repeat">						

							

							<table class="wrapper" width="100%">
                            	<thead>
        							<tr>
										<td width="10%" colspan="4"><span class="add btn-c green-btn">Add</span></td>
    								</tr>
								</thead>
                                <tbody class="container ui-sortable ">
                                <?php
								if ( $group1 ) :
								
									foreach ( $group1 as $field ) {
								?>
                                
									<tr class="row">
                                    	<td class="pointer" width="20%">
                                        	<span class="move blue-btn btn-c">Move</span>
                                        </td>
						
                                        <td width="70%">
                                        <label for="digital_cv_option1">Option1</label>
                                        	<dl>
        										
												<dt>option1</dt>
												<dd><input name="group1[$ic][0]" id="" type="text" class="form-control" value="<?php if($field[0] != '') echo esc_attr( $field[0] ); ?>" /></dd>
                                                <dt>option2</dt>
												<dd><input name="group1[$ic][1]" id="" type="text" class="form-control" value="<?php if($field[1] != '') echo esc_attr( $field[1] ); ?>" /> </dd>
                                                
        									</dl>
            							</td>
                                        <td class="pointer" width="10%">
                                        	<span class="remove red-btn btn-c">Remove</span>
                                        </td> 
									</tr>
                                    
                            <?php
							}
							else :
								// show a blank one
							?>
                                    <tr class="row">
                                    	<td class="pointer" width="20%">
                                        	<span class="move blue-btn btn-c">Move<label for="digital_cv_option1">Option1</label></span>
                                        </td>
										
						
                                        <td width="70%">
                                        	<dl>
        										
												<<dt>option1</dt>
												<dd><input name="group1[$ic][0]" id="" type="text" class="form-control" value="" /></dd>
                                                <dt>option2</dt>
												<dd><input name="group1[$ic][1]" id="" type="text" class="form-control" value="" /></dd>
        									</dl>
            							</td>
                                        <td class="pointer" width="10%">
                                        	<span class="remove red-btn btn-c">Remove</span>
                                        </td>
									</tr>
                            <?php endif; ?>	
                            		<tr class="template row" style="display: none;">
                                    	<td class="pointer" width="20%">
                                        	<span class="move blue-btn btn-c">Move<label for="digital_cv_option1">Option1</label></span>
                                        </td>
										
						
                                        <td width="70%">
                                        	<dl>
        										
												<dt>option1</dt>
												<dd><input name="group1[$ic][0]"  type="text" class="form-control" disabled/></dd>
                                                <dt>option2</dt>
												<dd><input name="group1[$ic][1]"  type="text" class="form-control" disabled/></dd>
        									</dl>
            							</td>
                                        <td class="pointer" width="10%">
                                        	<span class="remove red-btn btn-c">Remove</span>
                                        </td>
									</tr>	
                            	</tbody>							
							</table>

							<p>
								<input type="submit" class="save-btn btn-c" value="Submit" />
							</p>

							</div>
                            </form>`

Any suggestions on what I'm doing wrong?

**Comment by naifamoodi** [comment]
Hi,

Can you try changing group1[$ic] under <tr class="template row" style="display: none;"> to group1[{{row-count-placeholder}}]. Also place this template row over the prepopulated rows as opposed to under them.

**Comment by brad-klosterman** [comment]
Hi I changed both of these things but still after the rows are saved the rows load on page reload but when saved again they disappear in the database.  Well only the last value is saved so I am guessing they are all saved as one.

**Comment by naifamoodi** [comment]
Here you go: http://pastebin.com/RZRTehF3

**Comment by brad-klosterman** [comment]
I am running the jquery in a wordpress admin page menu. To display the menu page the php with the above code is called. I have enqueued the script admin_enqueue_scripts and saving is done with ajax. Is it something I am doing elsewhere outside of your code?

**Comment by naifamoodi** [comment]
group1[$ic] should be group1[<?php echo $ic; ?>]. The $ic variable also needs to be set. In your code it's not set.

**Comment by brad-klosterman** [comment]
In the jquery do I change {{row-count-placeholder}} to $ic

**Comment by brad-klosterman** [comment]
The template counter is working although when I generate the row the input field name is group1[0][1]

I used $ic in the Jquery 

How do I get the counter to work for the generated rows. 

`
<script type="text/javascript">
	jQuery(function() {
		jQuery('.repeat').each(function() {
    		jQuery(this).repeatable_fields();
    	});
    });
</script>

<script>
	jQuery(document).ready(function($){ 
 		$('#hide-show').click(function() {   $(this).next().slideToggle(1000); });
 	});
</script>





<script src="https://code.jquery.com/jquery-1.10.2.js"></script>




<body>
<?php
if(!empty($_POST)) {
    var_dump($_POST);
}
?>


    
<div class="userwrap">
	
	<h1>Work Experience</h1>
	
							<form id="pfile" action="" method="POST">	
                            <div class="repeat">						

							

							<table class="wrapper" width="100%">
                            	<thead>
        							<tr>
										<td width="10%" colspan="4"><span class="add btn-c green-btn">Add</span></td>
    								</tr>
								</thead>
                                <tbody class="container ui-sortable ">
                                
                                	<tr class="template row">
                                        <td  class="td-item" width="70%">
                                        
                                        		<span class="move blue-btn btn-c">Move</span>
                                      			
												<span class="profile-page-title">
                                                <input name="group1[$ic][0]" type="text" class="form-control"</span>
                                                <span id="hide-show" class="blue-btn btn-c">edit</span>
                                        		<dl id="obj-2" class="hidden">
												<dt>option1</dt>
												<dd><input name="group1[$ic][1]" type="text" class="form-control"/></dd>
                                                <dt>option2</dt>
												<dd><input name="group1[$ic][2]" type="text" class="form-control"/> </dd>
                                                
        										</dl>
            							</td>
                                        <td class="td-item" width="10%">
                                        	<span class="remove red-btn btn-c">Remove</span>
                                        </td> 
									</tr>
                                
								<?php
								
								
								
								
								if ( $group1 ) :
								$ic = 0;
								
									foreach ( $group1 as $field ) {
										
		
								?>
                                
                                 
                                
									<tr class="row">
                                        <td  class="td-item" width="70%">
                                        
                                        		<span class="move blue-btn btn-c">Move</span>
                                      			
												<span class="profile-page-title">
                                                <input name="group1[<?php echo $ic; ?>][0]" type="text" class="form-control" value="<?php if($field[0] != '') echo esc_attr( $field[0] ); ?>" /></span>
                                                <span id="hide-show" class="blue-btn btn-c">edit</span>
                                        		<dl id="obj-2" class="hidden">
												<dt>option1</dt>
												<dd><input name="group1[<?php echo $ic; ?>][1]" type="text" class="form-control" value="<?php if($field[1] != '') echo esc_attr( $field[1] ); ?>" /></dd>
                                                <dt>option2</dt>
												<dd><input name="group1[<?php echo $ic; ?>][2]" type="text" class="form-control" value="<?php if($field[2] != '') echo esc_attr( $field[2] ); ?>" /> </dd>
                                                
        										</dl>
            							</td>
                                        <td class="td-item" width="10%">
                                        	<span class="remove red-btn btn-c">Remove</span>
                                        </td> 
									</tr>
                                    
                            <?php
							}
							else :
								// show a blank one
							?>
                                    <tr class="row">
                                        <td  class="td-item" width="70%">
                                        
                                        		<span class="move blue-btn btn-c">Move</span>
                                      			
												<span class="profile-page-title">
                                                <input name="group1[<?php echo $ic; ?>][0]" type="text" class="form-control" /></span>
                                                <span id="hide-show" class="blue-btn btn-c">edit</span>
                                        		<dl id="obj-2" class="hidden">
												<dt>option1</dt>
												<dd><input name="group1[<?php echo $ic; ?>][1]" type="text" class="form-control" /></dd>
                                                <dt>option2</dt>
												<dd><input name="group1[<?php echo $ic; ?>][2]" type="text" class="form-control" /> </dd>
                                                
        										</dl>
            							</td>
                                        <td class="td-item" width="10%">
                                        	<span class="remove red-btn btn-c">Remove</span>
                                        </td> 
									</tr>
                            <?php endif; ?>	
                            			
                            	</tbody>							
							</table>

							<p>
								<input type="submit" class="save-btn btn-c" value="Submit" />
							</p>

							</div>
                            </form>
	
	
</div> 

</body>`

**Comment by brad-klosterman** [comment]
I have it displaying with

`$count = count( $group1 );
								for ( $ic = 0; $ic < $count; $ic++ )
								
								if ( $group1 ) :
									foreach ( $group1 as $field ) {`

the generated input fields display twice on page load..
